Source/Applications/openMIC/openMIC/FileMirroring/MirrorHandler.cs

```diff
@@ -159,7 +157,7 @@ protected string GetRemoteFilePath(string filePath)
             if (RemoteDirChar != "\\")
                 baseFilePath = baseFilePath.Replace("\\", RemoteDirChar);
 
-            return Path.Combine(GetRemotePath(), baseFilePath);
+            return CombineIntoPath(GetRemotePath(), baseFilePath);
```

Comment:
I thought Path.Combine works with either slash - docs on .NET says it does - but maybe that's Core only. Anyway there may be other gotchas with Path.Combine for use cases not considered.

Reply:
`Path.Combine()` is a bit quirky. Consider the following screenshot.

![image](https://user-images.githubusercontent.com/9203145/218005677-a92c8b58-fc69-4714-a49c-8737376cba41.png)

I was actually unaware of the first case until I ran this test. If `GetRemotePath()` already ends with `Path.AltDirectorySeparatorChar`, then it will keep the existing separator instead of replacing it with `Path.DirectorySeparatorChar`. To be fair, your code was indeed ensuring that `GetRemotePath()` ends with a forward slash so the second case _should_ be irrelevant.. although it doesn't help that it's a virtual function. However, the third case throws a big wrench in regardless. You'll end up having to trim leading backslashes from `GetBaseFilePath()` to avoid mangling the result with `Path.Combine()` at which point you may as well be using `string.Concat()`.

I suppose it's probably not realistic to consider systems where `Path.DirectorySeparatorChar` or `RemoteDirChar` is not one of the two slashes, but I've chosen to mention it anyway. `Path.Combine()` couldn't be made to be compatible with such a system.

As for whether there are other gotchas with `Path.Combine()`, I remember doing a lot of testing to exhaustively find all the quirks with leading and trailing slashes when I initially wrote `FtpPath.Combine()`. That function was necessary because the PQube devices would fail if you put in too many forward slashes so you could say it's been field tested. Obviously I'm not infallible, but I'm confident that it matches the behavior of `Path.Combine()` closely enough for this use case.

Reply:
In retrospect, maybe it should be renamed to `CombineIntoRemotePath()`.